pimd/pim_jp_agg.c

```diff
@@ -133,14 +134,22 @@ pim_jp_agg_get_interface_upstream_switch_list(struct pim_rpf *rpf)
 
 	for (ALL_LIST_ELEMENTS(pim_ifp->upstream_switch_list, node, nnode,
 			       pius)) {
-		if (pius->address.s_addr == rpf->rpf_addr.u.prefix4.s_addr)
-			break;
+#ifdef PIM_AF_IPV6
+		if (IPV6_ADDR_SAME(pius->address, rpf->rpf_addr.u.prefix4))
```

Comment:
We can use PIM_UN_PFX in place of rpf->rpf_addr.u.prefix4, then we don't need ifdef